Model/Order/DetailsCreator.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Swarming\SubscribePro\Model\Order;
+
+use Magento\Sales\Api\Data\OrderInterface;
+
+class DetailsCreator
+{
+    /**
+     * @var \Swarming\SubscribePro\Platform\Service\OrderDetails
+     */
+    private $orderDetailsService;
+
+    /**
+     * @var \SubscribePro\Service\OrderDetails\Builder
+     */
+    private $orderDetailsBuilder;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param \Swarming\SubscribePro\Platform\Service\OrderDetails $orderDetailsService
+     * @param \\SubscribePro\Service\OrderDetails\Builder $orderDetailsBuilder
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function __construct(
+        \Swarming\SubscribePro\Platform\Service\OrderDetails $orderDetailsService,
+        \SubscribePro\Service\OrderDetails\Builder $orderDetailsBuilder,
```

Comment:
Why is the orderDetailsBuilder located in SDK? The SDK should not know anything about Magneto order